server/src/main/java/io/druid/guice/BrokerIntermediateResultsPoolProvider.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.guice;
+
+import io.druid.java.util.common.ISE;
+import io.druid.query.DruidProcessingConfig;
+import io.druid.query.groupby.GroupByQueryConfig;
+import io.druid.query.groupby.strategy.GroupByStrategySelector;
+
+import javax.inject.Inject;
+
+@LazySingleton
+public class BrokerIntermediateResultsPoolProvider extends IntermediateResultsPoolProvider
```

Comment:
Likely this PR is going to be closed anyway, but as a side note, I find it much better to include words like "Historical" "Realtime", "Broker" into class names than not doing so. That is why I recently committed `HistoricalCursor` rather than `QueryableIndexStorageCursor`, (and yes, I would rename `QueryableIndexStorageAdapter` to `HistoricalAdapter`). The problem is that it is extremely hard to find the ends in the codebase, when you try to understand, what happens on a particular type of node. As far as I remember, I spent several hours first, when I was trying to understand where the core logic of Broker node is written (and right now I don't remember it!) Because classes like QueryResource, DirectDruidClient, AsyncQueryForwardingServlet, QueryableIndexStorageAdapter, have vague names and zero docs, so you can think that they may be part of any node type.